### DISCUSSION TITLE
Device MSC + dual CDC config

### DISCUSSION THREAD

**Discussion by bremenpl** [hathach/tinyusb#1930] [Q&A]
Hello there,
Based on the examples given in the repo I was able to configure a USB device with working MSC + CDC interfaces. What I am trying to do now is to change the configuration to have 1x MSC + 2x CDC interfaces. After adjusting the `usb_descriptors.c` none of the interfaces show up and I cannot figure out where did I make the mistake. Here is my `usb_descriptors.c` contents:

I am trying this config on an STM32F411 device (Device full speed support).
I would appreciate all help in resolving this one!

EDIT: Here are my config files:

[tusb_config_h.txt](https://github.com/hathach/tinyusb/files/10880603/tusb_config_h.txt)
[usb_descriptors_c.txt](https://github.com/hathach/tinyusb/files/10880607/usb_descriptors_c.txt)

For the above I had to change the extensions to TXT to be able to upload (from h and c consequently). Also I tidied up the `usb_descriptors` file a bit. 

Here is the boot up log from tinyusb with `LOG=2`:
[log.txt](https://github.com/hathach/tinyusb/files/10880615/log.txt)

**Comment by hathach** [comment]
- For readability please put your code as attached file or give a link to your github repo
- try to run make with `LOG=2` then post your log here as attached txt (if it is long enough).

**Reply by bremenpl** [reply]
Hi @hathach , thank you for the feedback. Here I attach:

The config files:
[tusb_config_h.txt](https://github.com/hathach/tinyusb/files/10880603/tusb_config_h.txt)
[usb_descriptors_c.txt](https://github.com/hathach/tinyusb/files/10880607/usb_descriptors_c.txt)

For the above I had to change the extensions to TXT to be able to upload (from h and c consequently). Also I tidied up the `usb_descriptors` file a bit. 

Here is the boot up log from tinyusb with `LOG=2`:
[log.txt](https://github.com/hathach/tinyusb/files/10880615/log.txt)

Starting from line 159 there are some failed assertions:

```
dcd_edpt_open 642: ASSERT FAILED
cdcd_open 305: ASSERT FAILED
process_set_config 956: ASSERT FAILED
process_control_request 706: ASSERT FAILED
  Stall EP0
USBD Bus Reset : Full Speed
```

But it's difficult for me to figure out what is this about exactly. I would appreciate your further feedback!

**Reply by hathach** [reply]
you also need to update your first post to remove the code block and put your attached descriptor there instead.

**Reply by hathach** [reply]
btw, the assert is pretty clear, the mcu does not have enough endpoints for your configuration. Check the stm32f411 manual, it only has 3 non-control endpoints. 

```C
  TU_ASSERT(epnum < ep_count);
```

**Reply by bremenpl** [reply]
> stm32f411

Hello,
Thank you a lot. I confirm with full honesty I did not see that coming for this MCU. I assumed it had at least 8 bidirectional end point. I will fix my original post.
